### PR TITLE
CHF-420 Add device health check to Bose DTH

### PR DESCRIPTION
**Pull request by marstorp**
Added health check to Bose DTH and updated connect app with Bose icons, fix for keeping the selected list during setup (submitOnChange: true), and syncing DTH with device after creation to update DTH tiles.

### CODE REVIEW

**Comment by marstorp** [comment]
@juano2310 can you code review this?
Tyler Gustafson have tested and given it an OK

File: devicetypes/smartthings/bose-soundtouch.src/bose-soundtouch.groovy [hunk 1]
```diff
@@ -837,6 +855,10 @@ def boseRefreshNowPlaying(delay=0) {
     return boseGET("/now_playing")
 }
 
+def boseSendGetNowPlaying() {
```
Comment (juano2310) [inline devicetypes/smartthings/bose-soundtouch.src/bose-soundtouch.groovy:858]: Is this related to Health Check?

**Review by juano2310 — APPROVED** [review]

**Comment by juano2310** [comment]
👍

File: devicetypes/smartthings/bose-soundtouch.src/bose-soundtouch.groovy (hunk 1 above)
Comment (marstorp) [reply devicetypes/smartthings/bose-soundtouch.src/bose-soundtouch.groovy:858]: Yes, it is, it's being setup by the cron job which is polling the device every two minutes, that drives the event that health check is using to see if everything is ok.
If there's events reported in the time configured health check will make a final call to poll() to see if device is around.
